status/reporter_gitlab.go

```diff
@@ -165,6 +167,13 @@ func (r *GitLabReporter) setCommitStatus(report TestReport) error {
 		}
 	}
 
+	// When trying to update the commit status to `enqueue` don't update
+	if existingCommitStatus != nil && glState == "enqueue" {
```

Comment:
I am afraid this `existingCommitStatus` is always `nil`, because it is declared [here](https://github.com/konflux-ci/integration-service/pull/853/files#diff-9173b9d26ca5486f2c07b5dfe966142cae4ca991f7787168bc520c2c655ac83aR56),  and then is set [here](https://github.com/konflux-ci/integration-service/pull/853/files#diff-9173b9d26ca5486f2c07b5dfe966142cae4ca991f7787168bc520c2c655ac83aR157-R162) only when `glState == gitlab.Running` .
So I feel you need to move `existingCommitStatus := r.GetExistingCommitStatus(allCommitStatuses, report.FullName)` to a line before [glState == gitlab.Running](https://github.com/konflux-ci/integration-service/pull/853/files#diff-9173b9d26ca5486f2c07b5dfe966142cae4ca991f7787168bc520c2c655ac83aR157)
Or  directly update [if glState == gitlab.Running](https://github.com/konflux-ci/integration-service/pull/853/files#diff-9173b9d26ca5486f2c07b5dfe966142cae4ca991f7787168bc520c2c655ac83aR157) to `if glState == gitlab.Running || glState == "enqueue"` and update their logs as well